std-bits/base/src/main/java/org/enso/base/Environment_Utils.java

```diff
@@ -14,30 +13,16 @@ public static String get_environment_variable(String name) {
     }
   }
 
-  /**
-   * Calls `action` with the provided environment variable.
-   *
-   * <p>The override is not persisted (its only visible from within the action called by this
-   * method) and it is only visible by the Enso `Environment.get` method (backed by {@code
-   * get_environment_variable}).
-   *
-   * <p>This is an internal function that should be used very carefully and only for testing.
-   */
-  public static <T> T with_environment_variable_override(
-      String name, String value, Function<Object, T> action) {
-    String oldValue = overrides.put(name, value);
-    boolean was_set = oldValue != null;
-    try {
-      // Giving 0 here as an argument, as using null would lead to incorrect behaviour, due to some
-      // weird Truffle peculiarity.
-      return action.apply(0);
-    } finally {
-      if (was_set) {
-        overrides.put(name, oldValue);
-      } else {
-        overrides.remove(name);
-      }
-    }
+  public static void setOverride(String name, String value) {
```

Comment:
How does one prevent this to be used in production?

Reply:
We cannot. Nor was this prevented in the earlier version.

We need some way to override environment variables in tests.

The functionality is hidden from the user. It does not do anything more than setting these env vars before launching Enso - so if the user wants to abuse the hidden feature, they are free to do so.